ext/drsyms/CMakeLists.txt

```diff
@@ -115,40 +115,41 @@ if (WIN32)
   set(dwarf_libpath "${dwarf_dir}/dwarf.lib")
   set(elftc_libpath "${dwarf_dir}/elftc.lib")
   configure_file("${dwarf_dir}/dwarf.pdb"
-    "${CMAKE_LIBRARY_OUTPUT_DIRECTORY}/dwarf.pdb" COPYONLY)
+          "${CMAKE_LIBRARY_OUTPUT_DIRECTORY}/dwarf.pdb" COPYONLY)
   configure_file("${dwarf_dir}/elftc.pdb"
-    "${CMAKE_LIBRARY_OUTPUT_DIRECTORY}/elftc.pdb" COPYONLY)
+          "${CMAKE_LIBRARY_OUTPUT_DIRECTORY}/elftc.pdb" COPYONLY)
   # while private loader means preferred base is not required, more efficient
   # to avoid rebase so we avoid conflict w/ client and other exts
   set(PREFERRED_BASE 0x76000000)
 
-elseif (UNIX)
+else ()
+  # Unix or Windows MSYS2/MinGW toolchain
   set(srcs
-    drsyms_unix_frontend.c drsyms_unix_common.c
-    drsyms_dwarf.c demangle.cc drsyms_common.c)
+          drsyms_unix_frontend.c drsyms_unix_common.c
+          drsyms_dwarf.c demangle.cc drsyms_common.c)
   if (APPLE)
     set(srcs ${srcs} drsyms_macho.c)
     set(dwarf_libpath
-      "${PROJECT_SOURCE_DIR}/ext/drsyms/libelftc-macho/lib${BITS}/libdwarf.a")
+            "${PROJECT_SOURCE_DIR}/ext/drsyms/libelftc-macho/lib${BITS}/libdwarf.a")
     set(elftc_libpath
-      "${PROJECT_SOURCE_DIR}/ext/drsyms/libelftc-macho/lib${BITS}/libelftc.a")
+            "${PROJECT_SOURCE_DIR}/ext/drsyms/libelftc-macho/lib${BITS}/libelftc.a")
   else (APPLE)
     set(srcs ${srcs} drsyms_elf.c)
     set(dwarf_libpath
-      "${PROJECT_SOURCE_DIR}/ext/drsyms/libelftc${ARCH}/lib${BITS}/libdwarf.a")
+            "${PROJECT_SOURCE_DIR}/ext/drsyms/libelftc${ARCH}/lib${BITS}/libdwarf.a")
```

Comment:
These are ELF binaries, so this is not correct for Windows MinGW.  Probably the MSVC-built libraries will work: since it's C code I would expect it to link with MinGW-compiled code.  The debug info in the .pdb files likely is not supported by MinGW though but we can live without that I guess.

Reply:
@derekbruening I presume user will not have MSVC toolchains at all. So we will need libelftc library build for MinGW. Are there any scripts which auto-build it?

Reply:
Like I said, the MSVC-built libelftc binaries are C, not C++, so I would expect them to link with MinGW-compiled code.  Is that not the case?  Did you try it?  If it fails I'm wondering what the reason is?

Reply:
@derekbruening First (and I think, it's enough) MinGW toolchain have not "lib.exe" or such.

Reply:
These are static PECOFF libraries.  You would link with them using the MinGW toolchain.  You do not need "lib.exe".  The only issues would be invalid interfaces (such as from C++ mangling scheme differences, but like I said this is C) or missing symbols (such as from MSVC security features or sthg: we could rebuild without them, or could make MinGW versions, though that takes up more space and complexity to have yet another version).

Reply:
I just tried linking to dwarf.lib with MinGW gcc and it does work, if these 3 symbols are resolved (I just made dummy functions to test the final link):

__security_check_cookie
__GSHandlerCheck
__chkstk

The first two can be eliminated by building /GS-.  We might have to provide our own probe for chkstk.  So using the existing PECOFF libraries can certainly be done.  However, given the chkstk especially, it might be cleaner to build MinGW versions of libeltc.